### DISCUSSION TITLE
Redirect doesn't work in mount

### DISCUSSION THREAD

**Discussion by natanaelsinisalo** [livewire/livewire#4266] [7. Bugs (archived)]
Hi,

In Livewire 2.6.* it used to be ok to do this:

```php
public function mount()
{
    if (!empty(request()->get('step', ''))) {
        return redirect()->route('example');
    }
}
```

Now after updating to 2.7.2 this gives:

Undefined offset: 1

And the file/line where it happens is:

`vendor/livewire/livewire/src/LivewireManager.php:399` (in the `getRootElementTagName` method)

It basically fails because it can't find any root element.

Is this a bug or have I just been using a feature that wasn't intended to be a feature? 🙈

**Comment by joshhanley** [comment]
@natanaelsinisalo we added a config option to allow people to choose whether render happens on redirect or not, but that should only impact `return $this->redirectRoute()` and default to not render (which is what you want).

Can you try using the above method and see if that works properly? If so, then I suspect there may be a bug with the changes when using the redirect helper.

Thanks!

**Reply by ghost** [reply]
I've got the same problem with:
```
public function mount()
{
    $this->redirectRoute('route.name');
}
```
The render method is skipped but the redirect don't run.

Livewire version `2.9.0`

**Comment by larylinz** [comment]
same problem... any solution? i'm on 2.10.5
